olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VolatileLayerClient.h

```diff
@@ -0,0 +1,119 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <memory>
+
+#include <olp/core/client/ApiError.h>
+#include <olp/core/client/ApiResponse.h>
+#include <olp/core/client/CancellationToken.h>
+#include <olp/core/client/HRN.h>
+#include <olp/core/client/OlpClientSettings.h>
+#include <olp/dataservice/read/DataRequest.h>
+#include <olp/dataservice/read/model/Data.h>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+class VolatileLayerClientImpl;
+
+/**
+ * @brief The VolatileLayerClient is aimed at acquiring data from a volatile
+ * layer of the OLP. The volatile layer is a key/value store. Values for a given
+ * key can change, and only the latest value is retrievable.
+ *
+ * Example:
+ * @code{.cpp}
+ * const std::string kCatalog = "hrn:here:data:::hereos-internal-test-v2";
+ * const std::string kLayerId = "hype-test";
+ * const auto kHRN = olp::client::HRN::FromString(kCatalog);
+ *
+ * std::shared_ptr<olp::thread::TaskScheduler> task_scheduler =
+ *    olp::client::OlpClientSettingsFactory::CreateDefaultTaskScheduler(1u);
+ * std::shared_ptr<olp::http::Network> http_client = olp::client::
+ *    OlpClientSettingsFactory::CreateDefaultNetworkRequestHandler();
+ *
+ * olp::authentication::Settings settings;
+ * settings.task_scheduler = task_scheduler;
+ * settings.network_request_handler = http_client;
+ *
+ * olp::client::AuthenticationSettings auth_settings;
+ * auth_settings.provider = olp::authentication::TokenProviderDefault(
+ *     kKeyId, kKeySecret, std::move(settings));
+ *
+ * auto client_settings = olp::client::OlpClientSettings();
+ * client_settings.authentication_settings = auth_settings;
+ * client_settings.task_scheduler = std::move(task_scheduler);
+ * client_settings.network_request_handler = std::move(http_client);
+ *
+ * VolatileLayerClient client{kHRN, kLayerId, client_settings};
+ * VolatileLayerClient::Callback<model::Data> callback = ... ;
+ * auto token = client.GetData(request, cb);
+ * @endcode
+ *
+ * @see
+ * https://developer.here.com/olp/documentation/data-api/data_dev_guide/shared_content/topics/olp/concepts/layers.html
+ */
+class DATASERVICE_READ_API VolatileLayerClient final {
+ public:
+  /// DataResult alias
+  using DataResult = model::Data;
+  /// ApiResponse template alias
+  template <typename Result>
+  using CallbackResponse = client::ApiResponse<Result, client::ApiError>;
+  /// Callback template alias
+  template <typename Response>
+  using Callback = std::function<void(CallbackResponse<Response> response)>;
```

Comment:
Consider to get rid of template parameter since only DataResult can be used as one.
These aliases meant to be used solely with VolatileLayerClient.

There is an idea to collapse all similar aliases to a single file and template them, but that will be a subject for further decisions.